cmake/OpenCVModule.cmake

```diff
@@ -573,6 +575,10 @@ macro(ocv_create_module)
     "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/cvconfig.h" "${OPENCV_CONFIG_FILE_INCLUDE_DIR}/opencv2/opencv_modules.hpp"
     ${${the_module}_pch})
 
+  set(OPENCV_MODULE_${the_module}_INTERFACE_DEFS ${OPENCV_MODULE_${the_module}_INTERFACE_DEFS}
+    CACHE INTERNAL "Public compile definitions of ${full_modname} module")
+  target_compile_definitions("${the_module}" PUBLIC ${OPENCV_MODULE_${the_module}_INTERFACE_DEFS})
```

Comment:
1) `target_compile_definitions` modifies "export(TARGET)" output and automatically adds these defines into `OpenCVModules.cmake`. Also it is propagated to depended targets (not only for targets in subdirectories).
2) `add_definitions` is fine for "interface" defines to build this project only
3) In general it should be avoided for "private" defines (to detect problems, like using of private defines in public headers)

BTW, are there any useful test/usage example of this feature? Are tests/samples built without any problems?